src/Treasure/Tui.hs

```diff
@@ -21,21 +24,34 @@ import qualified Graphics.Vty as V
 
 data MainBox
     = AccountsField Int
+    | TagsField Int
     deriving (Eq, Ord, Show)
 
+type TreasureField e = FormFieldState TreasureChest e MainBox
 
 -- Creates the inout form from the given initial 'TreasureChest'.
 mkForm :: forall e . TreasureChest -> Form TreasureChest e MainBox
-mkForm t = newForm (map makeCheckBox $ zip [0..] $ chestAccounts t) t
+mkForm t@TreasureChest{..} = newForm
+    ( toCheckBoxGroup "Accounts" chestAccountsL AccountsField chestAccounts
+   ++ toCheckBoxGroup "Tags"     chestTagsL     TagsField     chestTags
+    )
+    t
   where
-    makeCheckBox
-        :: (Int, CheckBox Account)
-        -> TreasureChest
-        -> FormFieldState TreasureChest e MainBox
-    makeCheckBox (i, CheckBox{..}) = checkboxField
-        (chestAccountsL . checkBoxL i)
-        (AccountsField i)
-        (prettyAccount checkboxData)
+    toCheckBoxGroup
+        :: forall a . Show a
+        => String
+        -> Lens' TreasureChest [CheckBox a]
```

Comment:
Aha, `Rank2Types` are needed because of `Lens`. At first, when I saw `Rank2Types` I thought that GHC suggests in a wrong way again and `ScopedTypeVariables` should be enough but here actually it's not enough. `Rank2Types` are needed because `Lens'` is a type alias for function that has `forall` inside.